lms/djangoapps/courseware/models.py

```diff
@@ -287,3 +289,118 @@ def course_allows_nonregistered_access(cls, course_id):
 
     def __unicode__(self):
         return u"{} : {} : {}".format(self.course_id, self.pref_key, self.pref_value)
+
+
+class GroupedQueries(models.Model):
+    """
+    Email Distribution List: Individual queries are associated with a grouped query.
+    A grouped query has a creation time, associated course and a name
+    """
+    title = models.CharField(max_length=255)
+    course_id = CourseKeyField(max_length=255, db_index=True)
+    created = models.DateTimeField(auto_now_add=True, null=True, db_index=True)
+
+    def __unicode__(self):
+        return "[GroupedQueries] Query {} for Course {}, {}".format(
+            self.id,  # pylint: disable=no-member
+            self.course_id,
+            self.title,
+        )
+
+INCLUSIONS = (
+    ('A', INCLUSION.AND),
+    ('N', INCLUSION.NOT),
+    ('O', INCLUSION.OR),
+)
+
+
+class QueriesSaved(models.Model):
```

Comment:
to elaborate, I think I like the model names to describe the python object, rather than the sql table or sql row.  Hence `SavedQuery` rather than `QueriesSaved`